src/charm.py

```diff
@@ -4,101 +4,424 @@
+        returns None.
         """
-        # Get a reference the container attribute on the PebbleReadyEvent
-        container = event.workload
-        # Define an initial Pebble layer configuration
-        pebble_layer = {
-            "summary": "httpbin layer",
-            "description": "pebble config layer for httpbin",
+        if "external_hostname" in self.model.config:
+            return self.model.config["external_hostname"]
+
+        namespace = self.model.name
+        # This could also be used
+        # with open("/var/run/secrets/kubernetes.io/serviceaccount/namespace", "r") as f:
+        #     namespace = f.read().strip()
+
+        client = Client()
+        traefik_service = client.get(Service, name=self.app.name, namespace=namespace)
+
+        if status := traefik_service.status:
+            if load_balancer_status := status.loadBalancer:
+                if ingress_addresses := load_balancer_status.ingress:
+                    ingress_address = ingress_addresses[0]
+                    return ingress_address.hostname or ingress_address.ip
+
+        return None
+
+    def _restart_traefik(self):
+        container = self.unit.get_container(_TRAEFIK_CONTAINER_NAME)
+
+        if not container.can_connect():
+            self.unit.status = WaitingStatus(
+                f"container '{_TRAEFIK_CONTAINER_NAME}' not yet ready"
+            )
+            return
+
+        updated_pebble_layer = {
+            "summary": "Traefik layer",
+            "description": "Pebble config layer for Traefik",
             "services": {
-                "httpbin": {
+                _TRAEFIK_SERVICE_NAME: {
                     "override": "replace",
-                    "summary": "httpbin",
-                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
-                    "startup": "enabled",
-                    "environment": {"thing": self.model.config["thing"]},
-                }
+                    "summary": "Traefik",
+                    "command": "/usr/local/bin/traefik",
+                },
             },
         }
-        # Add intial Pebble config layer using the Pebble API
-        container.add_layer("httpbin", pebble_layer, combine=True)
-        # Autostart any services that were defined with startup: enabled
-        container.autostart()
-        # Learn more about statuses in the SDK docs:
-        # https://juju.is/docs/sdk/constructs#heading--statuses
-        self.unit.status = ActiveStatus()
 
-    def _on_config_changed(self, _):
-        """Just an example to show how to deal with changed configuration.
+        current_pebble_layer = container.get_plan().to_dict()
 
-        TEMPLATE-TODO: change this example to suit your needs.
-        If you don't need to handle config, you can remove this method,
-        the hook created in __init__.py for it, the corresponding test,
-        and the config.py file.
+        if current_pebble_layer != updated_pebble_layer:
```

Comment:
```suggestion
        if current_pebble_layer["services"] != updated_pebble_layer["services"]:
```
?

Reply:
mmm, now that you mention it @jnsgruk , shouldn't we compare `current_pebble_layer["services"].to_dict()` and `updated_pebble_layer["services"].to_dict()` instead?

Reply:
Erm, you should be good I think

Reply:
Aaaactually, we do not have any variability in terms of CLI args. I think we are good to reduce a lot of complexity here. Brb.

Reply:
Done in 47c1ef3ff371d561c59c2ae5accda21efac1deb5